### PR TITLE
Reject orders that contain links

### PR DESCRIPTION
**Pull request by Ahmed-Ghanam**
## Description
Add a new extension method to detect URL(s) in order body using a Regex pattern.
Use this extension method to prevents usage of URLs in the email and SMS bodies.

## Related Issue(s)
- [573 Reject any order with links in texts](https://github.com/Altinn/altinn-notifications/issues/573)

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs [hunk 1]
```diff
@@ -230,4 +229,39 @@ public void IsValidEmail(string email, bool expectedResult)
...
+    [InlineData("You have received a new secure message. Please review it at http://mail.example.com.", false)]
+    [InlineData("Please verify your accounts: http://example.com/login and https://another-example.com.", false)]
+    [InlineData("For immediate action, visit http://192.168.1.1/security/update to secure your account.", false)]
+    [InlineData("Important notice: <a href='http://example.com/reset-password'>Reset your password here</a>.", false)]
+    [InlineData("You must log in to secure your account, visit our website at https://example.com to proceed.", false)]
+    [InlineData("Please verify your account at http://www.example[dot]com/login or contact us at support[at]example.com.", false)]
+    [InlineData("Alert: Unusual activity detected on your account. Log in at https://login.example-bank.com immediately.", false)]
+    [InlineData("Your package is waiting for delivery. Confirm your address here: https://secure.example.com/track/package12345.", false)]
+    [InlineData("Dear user, your account has been compromised. Please verify your details at http://example.com/login to avoid suspension.", false)]
```
Comment (acn-sbuad) [inline test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs:254]: You have many similar test cases here. Maybe limit it to a set of test cases that a clearly distinct? 
Some things that make sense to test are: 
- URL in begining of string
- URL mid string
- URL at end of string
- Varying url formats: ftp, http, https
- Encoded url

File: src/Altinn.Notifications/Validators/Rules/EmailTemplateRules.cs [hunk 2]
```diff
@@ -76,9 +77,17 @@ private static void ApplyCommonEmailTemplateRules<T>(IRuleBuilderInitial<T, Emai
                         .NotEmpty()
                         .WithMessage("The email template body must not be empty.");
 
+                    template.RuleFor(t => t.Body)
```
Comment (acn-sbuad) [inline src/Altinn.Notifications/Validators/Rules/EmailTemplateRules.cs:80]: it's possible to chain the rules if they pertain to the same property e.g.

```
template.RuleFor(t => t.Body)
	.NotEmpty()
	.WithMessage("The email template body must not be empty.")
	.Must(e => e.DoesNotContainUrl())
	.WithMessage("The email template body must not contain any URLs.");
```

I see pro's and cons with both approaches, so just wanted to mention it

Comment (Ahmed-Ghanam) [reply src/Altinn.Notifications/Validators/Rules/EmailTemplateRules.cs]: We use few validation rules, so chaining them together is ok.

File: test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs (hunk 1 above)
Comment (Ahmed-Ghanam) [reply test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs]: I moved the most of these test cases to their own class and minimized the number of test cases used with the email and SMS.

File: test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs [hunk 3]
```diff
@@ -230,4 +229,34 @@ public void IsValidEmail(string email, bool expectedResult)
         bool actual = RecipientRules.IsValidEmail(email);
         Assert.Equal(expectedResult, actual);
     }
+
+    [Theory]
+    [InlineData("Dear user, your account has been compromised. Please verify your details at http%3A%2F%2Fexample.com%2Flogin to avoid suspension.", false)]
+    public void Validate_BodyMustNotContainUrl(string emailBody, bool expectedResult)
```
Comment (acn-sbuad) [inline test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs:235]: nå er det på småååprik, men siden det bare er ett case kan man gå bort fra theory og inline data og heller bruke [Fact]

**Review by acn-sbuad — APPROVED** [review]
🎈

File: test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs (hunk 3 above)
Comment (Ahmed-Ghanam) [reply test/Altinn.Notifications.Tests/Notifications/TestingValidators/EmailNotificationOrderRequestValidatorTests.cs]: Replaced those three theories with facts.

**Pull request merged**